int32 gamma
---
int32 error